Wise_Owl_Library/Repositories/BookSqlRepository.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using Wise_Owl_Library.Models;
+using Wise_Owl_Library.Extensions;
+using Wise_Owl_Library.Models.Entities;
+
+namespace Wise_Owl_Library.Repositories
+{
+    public interface IBookRepository
+    {
+        Task<List<Book>> GetBooksAsync(string? title, int? stock);
+        Task<Book> GetBookAsync(int id);
+        Task<List<Book>> AddBooksAsync(List<Book> books);
+        Task<Book> UpdateBookAsync(Book book);
+        Task<bool> DeleteBookAsync(int id);
+        Task<bool> BookExistsAsync(string title, List<Author> authorNames);
+        Task<bool> BookExistsByIdAsync(int id);
+    }
+    public class BookSqlRepository(WiseOwlLibraryDbContext wiseOwlLibraryDbContext) : IBookRepository
+    {
+        public async Task<List<Book>> AddBooksAsync(List<Book> books)
+        {
+            List<Book> createdBooks = new();
+            foreach (Book book in books)
+            {
+                if(await wiseOwlLibraryDbContext.Books.AddAsync(book.ToBookEntity()) != null)
+                {
+                    createdBooks.Add(book);
+                }
+            }
+            await wiseOwlLibraryDbContext.SaveChangesAsync();
+
+            return createdBooks;
+        }
+
+        public async Task<bool> DeleteBookAsync(int id)
+        {
+            BookEntity? book = await wiseOwlLibraryDbContext.Books.FindAsync(id);
+            if (book == null)
+            {
+                throw new KeyNotFoundException($"Book with Id {id} not found.");
+            }
```

Comment:
To že tam není nemusí nutně znamenat chybu, záleží jestli to máš stateless nebo stateful